src/AutoRest.CSharp/Mgmt/Generation/ResourceGroupExtensionsWriter.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using AutoRest.CSharp.AutoRest.Plugins;
+using AutoRest.CSharp.Input;
+using AutoRest.CSharp.Mgmt.AutoRest;
+using AutoRest.CSharp.Mgmt.Decorator;
+using AutoRest.CSharp.Mgmt.Output;
+using AutoRest.CSharp.Output.Models;
+using AutoRest.CSharp.Output.Models.Types;
+using Azure.Core.Pipeline;
+using Azure.ResourceManager.Core;
+
+namespace AutoRest.CSharp.Generation.Writers
+{
+    internal class ResourceGroupExtensionsWriter
+    {
+        protected string Description = "A class to add extension methods to ResourceGroup.";
+        protected string Accessibility = "public";
+        protected string Type = "ResourceGroupExtensions";
+
+        public void WriteExtension(BuildContext<MgmtOutputLibrary> context, CodeWriter writer)
+        {
+            using (writer.Namespace(context.DefaultNamespace))
+            {
+                writer.WriteXmlDocumentationSummary(Description);
+                using (writer.Scope($"{Accessibility} static partial class {Type}"))
+                {
+                    foreach (var resource in context.Library.ArmResource)
+                    {
+                        if (resource.OperationGroup.Parent(context.Configuration.MgmtConfiguration).Equals("resourceGroups"))
+                        {
+                            foreach (var container in context.Library.ResourceContainers)
+                            {
+                                if (container.ResourceName == resource.Type.Name)
+                                {
+                                    writer.Line($"#region {resource.Type.Name}s");
+                                    WriteGetContainers(writer, resource, container);
+                                    writer.LineRaw("#endregion");
+                                    writer.Line();
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        private void WriteGetContainers(CodeWriter writer, Mgmt.Output.Resource armResource, ResourceContainer container)
+        {
+            // TODO: Find a solution to convert from single to plural
+            writer.WriteXmlDocumentationSummary($"Gets an object representing a {container.Type.Name} along with the instance operations that can be performed on it.");
+            writer.WriteXmlDocumentationParameter("resourceGroup", $"The <see cref=\"{typeof(ResourceGroupOperations)}\" /> instance the method will execute against.");
+            writer.WriteXmlDocumentation("return", $"Returns an <see cref=\"{container.Type.Name}\" /> object.");
```

Comment:
```suggestion
            writer.WriteXmlDocumentation("return", $"Returns an <see cref=\"{container.Type}\" /> object.");
```

Reply:
If we take this change, the code writer will add `\r\n` behind the type, not sure the why, maybe a issue? 
`"Returns a <see cref=\"global::Azure.ResourceManager.Sample.AvailabilitySetContainer\r\n\" /> object."`
The output will be:
```
/// <return>
/// Returns an <see cref="AvailabilitySetContainer
/// 
/// " /> object.
/// </return>
```

Reply:
Can you figure out where's the \r\n is coming from?

Reply:
`writer.Scope()`  method receives FormattedString as parameter, with Type, it will end in  [CodeWriter Line201](https://github.com/Azure/autorest.csharp/blob/f52c308d19dd54edbb3f440a6156a73923447429/src/AutoRest.CSharp/Common/Generation/Writers/CodeWriter.cs#L201), it append raw text **WITHOUT**  \r\t followed.
![image](https://user-images.githubusercontent.com/30464227/115170864-43822880-a0f4-11eb-9555-e36259623f61.png)

`writer.WriteXmlDocumentationXXXX()` method receives string type as parameter, so the CSharpType here will do its own convert, end in [CodeWriter Line439](https://github.com/Azure/autorest.csharp/blob/f52c308d19dd54edbb3f440a6156a73923447429/src/AutoRest.CSharp/Common/Generation/Writers/CodeWriter.cs#L439) , it append a new line ends with \r\t
![image](https://user-images.githubusercontent.com/30464227/115171638-0d45a880-a0f6-11eb-9487-8478b73e7012.png)

Reply:
Ugh. Yeah, the correct fix here would be to add support of FOrmattableStrings to the WriteXmlDocumentation*** but I don't want to block PR for that. `container.Type.Name` is a reasonable workaround.